programmers2/신규-아이디-추천.js

```diff
@@ -0,0 +1,14 @@
+function solution(new_id) {
+    new_id = new_id
+    .toLowerCase()
+    .replace(/[^\w-.)]/g, '')
```

Comment:
\w 가 word 를 표현하며 알파벳 + 숫자 + _ 중의 한 문자임을 의미한다. 
이런 의미를 가진다네요
주어진 1단계 조건을 보면 알파벳 숫자 _ 뿐만아니라 - 와 . 을 포함하여 테스트에 통과하지 못한것으로 추정됩니다. 
주어진 조건을 모두 충족하는 아래와 같은 방식으로 수정하면 채점까지 잘되네요!

```suggestion
    .replace(/[^-_.a-z0-9]/g, '')
```